js/modules/k6/ws/ws.go

```diff
@@ -125,6 +139,43 @@ func (*WS) Connect(ctx context.Context, url string, args ...goja.Value) (*WSHTTP
 				for _, key := range headersObj.Keys() {
 					header.Set(key, headersObj.Get(key).String())
 				}
+			case "cookies":
+				cookiesV := params.Get(k)
+				if goja.IsUndefined(cookiesV) || goja.IsNull(cookiesV) {
+					continue
+				}
+				cookies := cookiesV.ToObject(rt)
+				if cookies == nil {
+					continue
+				}
+				for _, key := range cookies.Keys() {
+					replace := false
+					value := ""
+					cookieV := cookies.Get(key)
+					if goja.IsUndefined(cookieV) || goja.IsNull(cookieV) {
+						continue
+					}
+					switch cookieV.ExportType() {
+					case reflect.TypeOf(map[string]interface{}{}):
+						cookie := cookieV.ToObject(rt)
+						for _, attr := range cookie.Keys() {
+							switch strings.ToLower(attr) {
+							case "replace":
+								replace = cookie.Get(attr).ToBoolean()
+							case "value":
+								value = cookie.Get(attr).String()
+							}
+						}
```

Comment:
I don't understand why this block is needed. You're parsing a cookie value as an object that can override the same cookie in the jar, but I'm not sure why this is necessary. Can you elaborate on your use case?